### PR TITLE
Firefly-1434: fixes: irsa 9, rubin-chart 2,3, empty FITS table

### PR DESCRIPTION
**Pull request by robyww**
#### [Firefly-1434](https://jira.ipac.caltech.edu/browse/FIREFLY-1434): multiple fixes

 - rubin chart 1: images and charts: arrows icon now material icon
 - rubin chart:2: images and charts: grouping display modes
 - irsa testing round 2: 9 - tap quoting
 - feedback: image viewer only show highlights when there is more then one showing
 - [Firefly-1435](https://jira.ipac.caltech.edu/browse/FIREFLY-1435): bug fix: reading an empty FITS table

#### Testing

 - https://fireflydev.ipac.caltech.edu/firefly-1434-several-fixes/firefly
 - fixes from feedback rubin chart 1 & 2, irsa testing round 2: 9
 - Upload a fits with an empty table HDU (example attached to ticket)
 - Image view will not show highlight when only on image or HiPS on screen

#### Confirmed fixed

- [x] rubin: chart: 1 : notes arrow are updated on both charts and images
- [x] rubin: chart: 2: note group implemented on both charts and images
- [x] irsa testing round 2: 9
- [x] firefly-1435
- [x] image viewer only highlights when multiple on screen

### CODE REVIEW

**Comment by lrebull** [comment]
oooh i have to say i really, really like the new arrows on scrolling between multiple plots and images. so nice! and I noticed the visual indicator of the single/multi plot or image mode right away. also, very nice and slick-looking.

**Comment by lrebull** [comment]
nearly everything else is ok!

i don’t understand this one, described variously as:

- feedback: image viewer only show highlights when there is more then one showing
- Image view will not show highlight when only on image or HiPS on screen
- image viewer only highlights when multiple on screen

i’ve been playing around with what i think this means, to no avail. the behavior is the same in your branch as it is on irsadev, and i’m thinking that that shouldn’t be the case if i’m reading this right…?

**Comment by lrebull** [comment]
ok now i understand the image highlighting! makes sense, and seems to be working.

File: src/firefly/java/edu/caltech/ipac/table/io/FITSTableReader.java [hunk 1]
```diff
@@ -244,6 +259,24 @@ private static Class<?> getClassType(Object base, int icol, boolean[] isScaled)
         }
     }
 
+    private static Class<?> getClassByTtype(String ttype) {
+        if (ttype==null) return String.class;
```
Comment (loitly) [inline src/firefly/java/edu/caltech/ipac/table/io/FITSTableReader.java:263]: This should be called ByTform and I think it will not recognize array types.

**Review by loitly — APPROVED** [review]

File: src/firefly/java/edu/caltech/ipac/table/io/FITSTableReader.java (hunk 1 above)
Comment (loitly) [reply src/firefly/java/edu/caltech/ipac/table/io/FITSTableReader.java:263]: I forgot to mention that "^[1-9]*" will not match any number with zero it them.  e.g. 10

Comment (robyww) [reply src/firefly/java/edu/caltech/ipac/table/io/FITSTableReader.java:263]: ok, I will clean it up.

**Pull request merged**